encfs/BlockNameIO.cpp

```diff
@@ -135,14 +135,17 @@ int BlockNameIO::maxDecodedNameLen(int encodedNameLen) const {
 }
 
 int BlockNameIO::encodeName(const char *plaintextName, int length, uint64_t *iv,
-                            char *encodedName) const {
+                            char *encodedName, int bufferLength) const {
+
   // copy the data into the encoding buffer..
+  rAssert(length <= (bufferLength - 2));
   memcpy(encodedName + 2, plaintextName, length);
 
   // Pad encryption buffer to block boundary..
   int padding = _bs - length % _bs;
   if (padding == 0) padding = _bs;  // padding a full extra block!
 
+  rAssert(padding <= (bufferLength - length - 2));
```

Comment:
Since "padding" isn't defined until after the first assert, should I rearrange the writes so that we write the padding first, and then write the data?